donate/payments/views.py

```diff
@@ -798,15 +801,18 @@ def form_valid(self, form, send_data_to_basket=True):
                 )
 
         elif send_data_to_basket:
-            data['source_url'] = self.request.build_absolute_uri()
-            data['lang'] = self.request.LANGUAGE_CODE
-            queue.enqueue(send_newsletter_subscription_to_basket, data)
-            queue_ga_event(self.request, ['send', 'event', {
-                    'eventCategory': 'Signup',
-                    'eventAction': 'Submitted the Form',
-                    'eventLabel': 'Email',
-                }
-            ])
+            if settings.DONATE_NEWSLETTER_SUBSCRIBE_METHOD.lower() == 'basket':
+                basket.subscribe(data['email'], 'mozilla-foundation', lang=self.request.LANGUAGE_CODE)
+            else:
+                data['source_url'] = self.request.build_absolute_uri()
+                data['lang'] = self.request.LANGUAGE_CODE
+                queue.enqueue(send_newsletter_subscription_to_basket, data)
+                queue_ga_event(self.request, ['send', 'event', {
```

Comment:
TODO: This needs to be moved outside the else statement. Always trigger a GA event.

Reply:
If that still needs to happen, R- until that's done =)

Reply:
Done.